lib/models/response_state.dart

```diff
@@ -0,0 +1,12 @@
+abstract class ResponseState<T> {}
```

Comment:
@shubhamsinghmutualmobile will it be better if we call this UiState ?

Reply:
Yes, that would be better
Let me make the change real quick 👍